packages/node_modules/@ciscospark/storage-adapter-session-storage/src/index.js

```diff
@@ -0,0 +1,147 @@
+     */
+    this.Bound = class {
+      /**
+       * @constructs {Bound}
+       * @param {string} namespace
+       * @param {Object} options
+       */
+      constructor(namespace, options) {
+        namespaces.set(this, namespace);
+        loggers.set(this, options.logger);
+      }
+
+      /**
+       * @private
+       * @returns {rawData}
+       */
+      _getRawData() {
+        const rawData = sessionStorage.getItem(basekey);
+        return rawData ? JSON.parse(rawData) : {};
+      }
+
+      /**
+       * @private
+       * @returns {mixed}
+       */
+      _load() {
+        const allData = this._getRawData();
+        return allData[namespaces.get(this)] || {};
+      }
+
+      /**
+       * @param {Object} data
+       * @private
+       * @returns {undefined}
+       */
+      _save(data) {
+        const allData = this._getRawData();
+        allData[namespaces.get(this)] = data;
+
+        sessionStorage.setItem(basekey, JSON.stringify(allData));
+      }
+
+      /**
+       * Clears the sessionStorage
+       * @param {string} key
+       * @returns {Promise}
+       */
+      clear() {
+        loggers.get(this).info('session-storage-store-adapter: clearing sessionStorage');
+        return Promise.resolve(sessionStorage.removeItem(basekey));
+      }
+
+      /**
+       * Removes the specified key
+       * @param {string} key
+       * @returns {Promise}
+       */
+      del(key) {
+        return new Promise((resolve) => {
+          loggers.get(this).info(`session-storage-store-adapter: deleting \`${key}\``);
+          const data = this._load();
+          Reflect.deleteProperty(data, key);
+          this._save(data);
+          resolve();
+        });
+      }
+
+      /**
+       * Retrieves the data at the specified key
+       * @param {string} key
+       * @returns {Promise<mixed>}
+       */
+      get(key) {
+        return new Promise((resolve, reject) => {
+          loggers.get(this).info(`session-storage-store-adapter: reading \`${key}\``);
+          const data = this._load();
+          const value = data[key];
+          if (typeof value !== 'undefined') {
+            return resolve(value);
+          }
+
+          return reject(new NotFoundError(`No value found for ${key}`));
+        });
+      }
+
+      /**
+       * Stores the specified value at the specified key
+       * @param {string} key
+       * @param {mixed} value
+       * @returns {Promise}
+       */
+      put(key, value) {
+        return new Promise((resolve) => {
+          loggers.get(this).info(`session-storage-store-adapter: writing \`${key}\``);
+          const data = this._load();
+          data[key] = value;
+          this._save(data);
+          resolve();
+        });
+      }
+    };
+  }
+
+  /**
+   * Returns an adapter bound to the specified namespace
+   * @param {string} namespace
+   * @param {Object} options
+   * @returns {Promise<Bound>}
+   */
+  bind(namespace, options) {
```

Comment:
probably not in scope for this PR, but i think we need to fallback to the memory store adapter if session storage or local storage doesn't existing (like in private safari)